From f23cbade5ecc451b9a01bd11dd9de7456424722c Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Mon, 1 Jun 2020 17:02:08 +0000
Subject: [PATCH] GCC: DOMRect constexpr equal operator depends on non
 constexpr operators

Make operators 'inline' only.

Bug: 819294
Change-Id: If07442258b4ebce26c013c4dff830c1d61dff9e3
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2217415
Commit-Queue: Fernando Serboncini <fserb@chromium.org>
Reviewed-by: Fernando Serboncini <fserb@chromium.org>
Cr-Commit-Position: refs/heads/master@{#773698}
---
 third_party/blink/renderer/core/geometry/dom_rect.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/third_party/blink/renderer/core/geometry/dom_rect.h b/third_party/blink/renderer/core/geometry/dom_rect.h
index fede00578e6b..058c60a904ef 100644
--- a/third_party/blink/renderer/core/geometry/dom_rect.h
+++ b/third_party/blink/renderer/core/geometry/dom_rect.h
@@ -34,11 +34,11 @@ class CORE_EXPORT DOMRect final : public DOMRectReadOnly {
   void setHeight(double height) { height_ = height; }
 };
 
-constexpr bool operator==(const DOMRect& lhs, const DOMRect& rhs) {
+inline bool operator==(const DOMRect& lhs, const DOMRect& rhs) {
   return lhs.x() == rhs.x() && lhs.y() == rhs.y() &&
          lhs.width() == rhs.width() && lhs.height() == rhs.height();
 }
-constexpr bool operator!=(const DOMRect& lhs, const DOMRect& rhs) {
+inline bool operator!=(const DOMRect& lhs, const DOMRect& rhs) {
   return !(lhs == rhs);
 }
 }  // namespace blink
-- 
2.26.2

